systemd/systemd.go

```diff
@@ -1392,22 +1400,41 @@ const (
 	snappyOriginModule = "X-SnapdOrigin"
 )
 
-func writeMountUnitFile(u *MountUnitOptions) (mountUnitName string, err error) {
+func ensureMountUnitFile(u *MountUnitOptions) (mountUnitName string, modified MountUpdateStatus, err error) {
 	if u == nil {
-		return "", errors.New("writeMountUnitFile() expects valid mount options")
+		return "", MountUnchanged, errors.New("ensureMountUnitFile() expects valid mount options")
 	}
 
 	mu := MountUnitPathWithLifetime(u.Lifetime, u.Where)
 	outf, err := osutil.NewAtomicFile(mu, 0644, 0, osutil.NoChown, osutil.NoChown)
 	if err != nil {
-		return "", fmt.Errorf("cannot open mount unit file: %v", err)
+		return "", MountUnchanged, fmt.Errorf("cannot open mount unit file: %v", err)
 	}
 	defer outf.Cancel()
 
 	if err := parsedMountUnitTemplate.Execute(outf, &u); err != nil {
-		return "", fmt.Errorf("cannot generate mount unit: %v", err)
+		return "", MountUnchanged, fmt.Errorf("cannot generate mount unit: %v", err)
+	}
+
+	f, err := os.Open(mu)
+	defer f.Close()
+	if os.IsNotExist(err) {
+		modified = MountCreated
+	} else {
+		if err != nil {
```

Comment:
I wonder if instead of this code we shouldn't move this code to use osutil.EnsureFileState. that one doesn't what kind of changes it made but it could be made to do so, as other functions in its family do that

Reply:
It now uses `osutil.EnsureFileState`.